policy-test/src/grpc.rs

```diff
@@ -0,0 +1,164 @@
+//! A gRPC client for the inbound policy API.
+//!
+//! This client currently discovers a destination controller pod via the k8s API and uses port
+//! forwarding to connect to a running instance.
+
+use anyhow::Result;
+use linkerd2_proxy_api::inbound::inbound_server_policies_client::InboundServerPoliciesClient;
+pub use linkerd2_proxy_api::*;
+use linkerd_policy_controller_k8s_api::{self as k8s, ResourceExt};
+use tokio::io;
+
+#[macro_export]
+macro_rules! assert_is_default_deny {
+    ($config:expr) => {
+        assert_eq!(
+            $config.labels,
+            Some(("name".to_string(), "default:deny".to_string()))
+                .into_iter()
+                .collect()
+        );
+        assert!($config.authorizations.is_empty());
+    };
+}
+
+#[macro_export]
+macro_rules! assert_protocol_detect {
+    ($config:expr) => {{
+        use linkerd2_proxy_api::inbound;
+
+        assert_eq!(
+            $config.protocol,
+            Some(inbound::ProxyProtocol {
+                kind: Some(inbound::proxy_protocol::Kind::Detect(
+                    inbound::proxy_protocol::Detect {
+                        timeout: Some(time::Duration::from_secs(10).into()),
+                    }
+                )),
+            }),
+        );
+    }};
+}
+
+#[derive(Debug)]
+pub struct PolicyClient {
+    client: InboundServerPoliciesClient<GrpcHttp>,
+}
+
+#[derive(Debug)]
+struct GrpcHttp {
+    tx: hyper::client::conn::SendRequest<tonic::body::BoxBody>,
+}
+
+// === impl PolicyClient ===
+
+impl PolicyClient {
+    pub async fn port_forwarded(client: &kube::Client) -> Result<Self> {
+        let pod = Self::get_policy_controller_pod(client).await?;
+        let io = Self::connect_port_forward(client, &pod).await?;
+        let http = GrpcHttp::handshake(io).await?;
+        Ok(PolicyClient {
+            client: InboundServerPoliciesClient::new(http),
+        })
+    }
```

Comment:
i wonder if we want to stick [`anyhow`'s `Context`s](https://docs.rs/anyhow/latest/anyhow/trait.Context.html) on some of these errors so that when we unwrap them or whatever, we get more useful information about what went wrong?
```suggestion
    pub async fn port_forwarded(client: &kube::Client) -> Result<Self> {
        let pod = Self::get_policy_controller_pod(client)
            .await
            .context("getting policy controller pod failed")?;
        let io = Self::connect_port_forward(client, &pod)
            .await
            .context("connecting port-forward failed")?;
        let http = GrpcHttp::handshake(io)
            .await
            .context("gRPC handshake with policy controller failed")?;
        Ok(PolicyClient {
            client: InboundServerPoliciesClient::new(http),
        })
    }
```

Reply:
(we could do something similar for the other functions here)